common/src/androidTest/java/com/microsoft/identity/common/CommandDispatcherTest.java

```diff
@@ -0,0 +1,298 @@
+
+        final BaseCommand<String> testCommand = getTestCommand(testLatch);
+        CommandDispatcher.submitSilent(testCommand);
+        testLatch.await();
+    }
+
+    private TestCommand getTestCommand(final CountDownLatch testLatch) {
+        return new TestCommand(
+                getEmptyTestParams(),
+                new CommandCallback<String, Exception>() {
+                    @Override
+                    public void onCancel() {
+                        testLatch.countDown();
+                        Assert.fail();
+                    }
+
+                    @Override
+                    public void onError(Exception error) {
+                        testLatch.countDown();
+                        Assert.fail();
+                    }
+
+                    @Override
+                    public void onTaskCompleted(String s) {
+                        testLatch.countDown();
+                        Assert.assertEquals(TEST_RESULT_STR, s);
+                    }
+                }, 0);
+    }
+
+    /**
+     * This test represents the case where a command changes underneath our system
+     * while we're using it as a key.  They're not immutable, so they're not safe to
+     * use as keys in a map.  It won't hurt, though, unless we can't get rid of them.
+     * To test this, we submit a command, block before it executes, alter it, release it,
+     * and then make sure it gets cleaned up.
+     * @throws Exception
+     */
+    @Test
+    public void testSubmitSilentWithParamMutation() throws Exception {
+        final CountDownLatch testLatch = new CountDownLatch(1);
+        CountDownLatch submitLatch = new CountDownLatch(1);
+        CountDownLatch submitLatch1 = new CountDownLatch(1);
+
+        final TestCommand testCommand = new LatchedTestCommand(
+                getEmptyTestParams(),
+                new CommandCallback<String, Exception>() {
+                    @Override
+                    public void onCancel() {
+                        testLatch.countDown();
+                        Assert.fail();
+                    }
+
+                    @Override
+                    public void onError(Exception error) {
+                        testLatch.countDown();
+                        Assert.fail();
+                    }
+
+                    @Override
+                    public void onTaskCompleted(String s) {
+                        testLatch.countDown();
+                        Assert.assertEquals(TEST_RESULT_STR, s);
+                    }
+                }, 0, submitLatch, submitLatch1);
+        FinalizableResultFuture<CommandResult> f = CommandDispatcher.submitSilentReturningFuture(testCommand);
+        submitLatch1.await();
+        testCommand.value = 2;
+        submitLatch.countDown();
+        testLatch.await();
+        Assert.assertTrue(f.isDone());
+        Assert.assertEquals(TEST_RESULT_STR, f.get().getResult());
+        f.isCleanedUp();
```

Comment:
This looks like something we can use RoboTestUtils.flushScheduler() for?

(no need to make changes in this PR)

Reply:
Hah.  Yeah, I could have just gotten a reference on the executor and shut it down, I suppose.

Reply:
could we add a TODO to migrate to that one (and get rid of the new future class) at some point in the future? (e.g. when you're working on the root cause fix)